auth/src/main/kotlin/com/hedvig/android/auth/network/LoginStatusResponse.kt

```diff
@@ -0,0 +1,42 @@
+package com.hedvig.android.auth.network
+
+import com.hedvig.android.auth.AuthorizationCode
+import com.hedvig.android.auth.LoginStatusResult
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Response
+
+@Serializable
+data class LoginStatusResponse(
+  val status: LoginStatus,
+  val seBankidHintCode: String?,
+  val authorizationCode: String?,
+) {
+  enum class LoginStatus {
+    PENDING, FAILED, COMPLETED
+  }
+}
+
+fun Response.toLoginStatusResult(): LoginStatusResult {
+  val responseBody = body?.string()
+  val result = if (isSuccessful && responseBody != null) {
+    Json.decodeFromString<LoginStatusResponse>(responseBody).toLoginStatusResult()
+  } else {
+    LoginStatusResult.Failed(message = message)
+  }
+  return result
+}
```

Comment:
Hmm what do you think about putting these functions inside the file where it was used?
Otherwise I'd imagine we'd start getting quite a namespace pollution on `Response`, getting a lot of not always relevant extension functions on it in places of the module where it wouldn't make a lot of sense.
But not sure how exactly it'd look like and if we do want access to this function in other places, so giving the idea here to get feedback.